gcc 13 warns about extra std::move() calls

```
kdalgorithms-src/src/bits/find_if.h:135:43: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
  135 |     return detail::create_result<Iterator>(std::move(it), std::move(range.begin()),
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  136 |                                            std::move(range.end()),
      |                                            ~~~~~~~~~~~~~~~~~~~~~~~
  137 |                                            std::is_lvalue_reference<Container>());
      |                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
kdalgorithms-src/src/bits/find_if.h:135:43: note: remove ‘std::move’ call
kdalgorithms-src/src/bits/find_if.h:135:43: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
kdalgorithms-src/src/bits/find_if.h:135:43: note: remove ‘std::move’ call
```
The code is quite standard I think:
```
			const auto &objects = aMethodThatReturnsAConstRefToAQList();
			auto object = kdalgorithms::find_if(objects, [product](object) { return object->product() == product; });
```

--- a/src/bits/find_if.h
+++ b/src/bits/find_if.h
@@ -132,8 +132,8 @@ auto find_if(Container &&container, UnaryPredicate &&predicate)
     auto it = std::find_if(range.begin(), range.end(),
                            detail::to_function_object(std::forward<UnaryPredicate>(predicate)));
     using Iterator = decltype(range.begin());
-    return detail::create_result<Iterator>(std::move(it), std::move(range.begin()),
-                                           std::move(range.end()),
+    return detail::create_result<Iterator>(std::move(it), range.begin(),
+                                           range.end(),
                                            std::is_lvalue_reference<Container>());
 }
 

